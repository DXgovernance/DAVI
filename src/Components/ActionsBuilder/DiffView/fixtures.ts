export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
