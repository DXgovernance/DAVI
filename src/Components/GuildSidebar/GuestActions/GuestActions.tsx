import { Button } from 'old-components/Guilds/common/Button';

export interface GuestActionsProps {
  userWalletAddress?: string;
  onShowStakeModal?: () => void;
  onShowWalletModal?: () => void;
}

export const GuestActions: React.FC<GuestActionsProps> = ({
  userWalletAddress,
  onShowStakeModal,
  onShowWalletModal,
}) => {
  return (
    <>
      {userWalletAddress ? (
        <Button
          data-testid="open-stake-tokens-modal-btn"
          variant="secondary"
          onClick={onShowStakeModal}
        >
          Join
        </Button>
      ) : (
        <Button variant="secondary" onClick={onShowWalletModal}>
          Connect Wallet
        </Button>
      )}
    </>
  );
};
