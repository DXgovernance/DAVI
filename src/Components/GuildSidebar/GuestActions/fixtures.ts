import { GuestActionsProps } from './GuestActions';

export const withData: GuestActionsProps = {
  userWalletAddress: '0x0000000000000000000000000000000000000000',
  onShowStakeModal: jest.fn(),
  onShowWalletModal: jest.fn(),
};
