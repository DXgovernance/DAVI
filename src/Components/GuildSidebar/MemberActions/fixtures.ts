import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberActionsProps } from './MemberActions';

export const propsWithData: MemberActionsProps = {
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  userWalletAddress: '0x1234567890123456789012345678901234567890',
  userEnsAvatar: {
    imageUrl: '/assets/favicon-32x32.png',
    ensName: 'test.eth',
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
};
