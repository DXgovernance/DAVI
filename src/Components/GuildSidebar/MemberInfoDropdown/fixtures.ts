import { BigNumber } from 'ethers';
import moment from 'moment';
import { MemberInfoDropdownProps } from './MemberInfoDropdown';

export const closedWithoutData: MemberInfoDropdownProps = {
  isOpen: false,
  onClose: jest.fn(),
};

export const openWithoutData: MemberInfoDropdownProps = {
  isOpen: true,
  onClose: jest.fn(),
};

export const closedWithData: MemberInfoDropdownProps = {
  isOpen: false,
  onClose: jest.fn(),
  guildToken: {
    decimals: 18,
    name: 'Guild Token',
    symbol: 'GT',
    totalSupply: BigNumber.from('1000000000'),
  },
  isRepGuild: false,
  unlockedAt: moment().add(1, 'day'),
  userVotingPower: BigNumber.from('250000000'),
  userVotingPowerPercent: 50,
  onShowStakeModal: jest.fn(),
  onWithdraw: jest.fn(),
};

export const openWithData: MemberInfoDropdownProps = {
  ...closedWithData,
  isOpen: true,
};
