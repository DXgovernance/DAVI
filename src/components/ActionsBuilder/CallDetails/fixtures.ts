import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
