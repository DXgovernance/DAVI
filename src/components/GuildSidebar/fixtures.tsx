import { GuildSidebarProps } from './GuildSidebar';

const TestDiv = () => <div>Test</div>;

export const withData: GuildSidebarProps = {
  guildName: 'Test Guild',
  numberOfMembers: 10,
  actions: TestDiv(),
};
