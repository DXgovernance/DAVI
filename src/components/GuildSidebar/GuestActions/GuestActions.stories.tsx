import { ComponentStory, ComponentMeta } from '@storybook/react';
import { withData } from './fixtures';
import { GuestActions } from './GuestActions';

export default {
  title: 'Sidebars/GuildSidebar/GuestActions',
  component: GuestActions,
} as ComponentMeta<typeof GuestActions>;

const Template: ComponentStory<typeof GuestActions> = args => (
  <GuestActions {...args} />
);

export const HasWallet = Template.bind({});
HasWallet.args = withData;

export const NoWallet = Template.bind({});
NoWallet.args = {};
