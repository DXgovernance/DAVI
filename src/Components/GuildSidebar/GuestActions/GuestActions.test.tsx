import { GuestActions } from './GuestActions';
import { render } from 'utils/tests';
import { withData } from './fixtures';

describe('GuestActions', () => {
  it('Should match snapshot without data', () => {
    const { container } = render(<GuestActions />);
    expect(container).toMatchSnapshot();
  });

  it('Should match snapshot with data', () => {
    const { container } = render(<GuestActions {...withData} />);
    expect(container).toMatchSnapshot();
  });
});
