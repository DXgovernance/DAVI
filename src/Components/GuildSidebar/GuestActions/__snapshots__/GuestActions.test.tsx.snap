// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestActions Should match snapshot with data 1`] = `
<div>
   
  <button
    class="sc-bdfBQB cAORUT"
    data-testid="open-stake-tokens-modal-btn"
  >
    Join
  </button>
   
</div>
`;

exports[`GuestActions Should match snapshot without data 1`] = `
<div>
   
  <button
    class="sc-bdfBQB cAORUT"
  >
    Connect Wallet
  </button>
   
</div>
`;
